### PR TITLE
Add device code flow support to MSAL Test app

### PR DESCRIPTION
**Pull request by rpdome**
- Added a button for triggering DCF.
- Added a "Status" section which displays the latest toast message.

Screen Shot 2020-08-24 at 6 13 08 PM

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ void acquireToken(@NonNull final Activity activity,
      * @param scopes the desired access scopes
      * @param callback callback object used to communicate with the API throughout the protocol
      */
-    void acquireTokenWithDeviceCode(@Nullable String[] scopes, @NonNull final DeviceCodeFlowCallback callback);
+    void acquireTokenWithDeviceCode(@NonNull String[] scopes, @NonNull final DeviceCodeFlowCallback callback);
```
Comment (shahzaibj) [inline msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java:94]: 👍

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalWrapper.java [hunk 2]
```diff
@@ -193,6 +194,33 @@ public void acquireTokenSilentWithResource(@NonNull RequestOptions requestOption
 
     abstract void acquireTokenSilentAsyncInternal(@NonNull final AcquireTokenSilentParameters parameters);
 
+    public void acquireTokenSilentWithDeviceCodeFlow(@NonNull RequestOptions requestOptions,
```
Comment (shahzaibj) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalWrapper.java:197]: nit: rename to `acquireTokenWithDeviceCodeFlow`

**Comment by shahzaibj** [comment]
So status is NOT displaying a toast, and rather displaying in the TextView, right?

Also what does the Copy button copy? The user code or the uri or both?

**Comment by rpdome** [comment]
> So status is NOT displaying a toast, and rather displaying in the TextView, right?
> 
> Also what does the Copy button copy? The user code or the uri or both?

I'm keeping the toast. The TextView will hold the messages that is displayed by the toast. 
The copy button will copy the whole message. (So now you can copy both code and URI in this scenario, or error code/message when an error is returned).

**Pull request merged**